### PR TITLE
Prometheus stack/o11y stack

### PR DESCRIPTION
**Pull request by rounakupadhye**
Added dashboard automation part and additional datasource part in prometheus-values.yaml also added dashboard.yaml file with dashboards json.

### CODE REVIEW

File: monitoring/dashboard.json [hunk 1]
```diff
@@ -0,0 +1,4553 @@
+apiVersion: v1
```
Comment (Hashfyre) [inline monitoring/dashboard.json]: This is configMap YAML which should have .yml /. yaml extension.
We can keep the JSON as a separate file and load in using helm templating

```
data:
  {{-/* Note, indent of only two spaces */}}
  config.json: |-
{{ .Files.Get "my-config.json" | indent 4 }}
{{/* .Get, not .Glob; indent 4 spaces, more than 2 above */}}
```

Comment (Hashfyre) [reply monitoring/dashboard.json]: ref: https://stackoverflow.com/questions/64662568/how-can-i-use-a-json-file-in-my-configmap-yaml-helm

**Review by Hashfyre — CHANGES_REQUESTED** [review]

File: monitoring/dashboard.json (hunk 1 above)
Comment (rounakupadhye) [reply monitoring/dashboard.json]: Updated the file extension from dashboard.json to dashboard.yaml

**Review by DirDave — APPROVED** [review]
GTG

**Review by Hashfyre — APPROVED** [review]
LGTM

**Pull request merged**